src/zcbenchmarks.cpp

```diff
@@ -88,3 +90,15 @@ double benchmark_create_joinsplit()
     assert(pourtx.Verify(*pzerocashParams));
     return ret;
 }
+
+double benchmark_solve_equihash()
+{
+    const char *testing = "testing";
```

Comment:
Hm. We want to see how long the *average nonce try* takes, so in that light I think you're right that testing until we get a solution is incorrect, but I think we should act over a random input, and benchmark it several times to get a solid average.

```c++
CBlock pblock;
CEquihashInput I{pblock};
CDataStream ss(SER_NETWORK, PROTOCOL_VERSION);
ss << I;

Equihash eh {Params(CBaseChainParams::MAIN).EquihashN(), Params(CBaseChainParams::MAIN).EquihashK()};
crypto_generichash_blake2b_state eh_state;
eh.InitialiseState(eh_state);
crypto_generichash_blake2b_update(&eh_state, (unsigned char*)&ss[0], ss.size());

uint256 nonce;
randombytes_buf(nonce.begin(), 32); // bring in sodium.h for this
crypto_generichash_blake2b_update(&eh_state,
                                  nonce.begin(),
                                  nonce.size());

eh.BasicSolve(curr_state);
```